src/main/java/com/iota/iri/service/transactionpruning/async/AsyncTransactionPruner.java

```diff
@@ -67,6 +68,11 @@
      */
     private SpentAddressesService spentAddressesService;
 
+    /**
+     * Used to check whether an address is already persisted in the persistence layer.
+     */
```

Comment:
I just want to let you know my opinion.
These types of comments could have simply been placed only on the `SpentAddressProvider` class file and that's it. Via an IDE you could read it from all classes that use it as dependency.

Right now putting such a comment in every place we use the `SpentAddressProvider` is kind of repetitive, and pretty much violates DRY (do not repeat yourself). If we want to change the comment we have to change it in multiple places.

The reason we have those is because Hans, who did a great job, really liked to have them everywhere.

Having said that, since currently the entire class looks like this, it is perfectly ok that you added the comment and no need to remove it now.